### PR TITLE
Handle missing field in error response

### PR DESCRIPTION
**Pull request by larkox**
#### Summary
When adding a field submission response had a error that did not map to any field, it considered as if no errors occurred and closed. Now we are showing a general error specifying the field and the error.

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-34640

#### Related Pull Requests
Mobile: https://github.com/mattermost/mattermost-mobile/pull/5461

#### Release Note
```release-note
NONE
```

### CODE REVIEW

File: components/apps_form/apps_form.tsx [hunk 1]
```diff
@@ -95,12 +95,20 @@ export class AppsForm extends React.PureComponent<Props, State> {
...
+        if (fieldErrors && Object.keys(fieldErrors).length >= 0) {
             hasErrors = true;
-            state.fieldErrors = fieldErrors;
+            if (checkIfErrorsMatchElements(fieldErrors as any, elements)) {
+                state.fieldErrors = fieldErrors;
+            } else if (!state.formError) {
+                const field = Object.keys(fieldErrors)[0];
+                state.formError = this.props.intl.formatMessage({
+                    id: 'apps.error.responses.unknown_field_error',
+                    defaultMessage: 'Received an error for an unkown field. Field name: `{field}`. Error: `{error}`.',
```
Comment (catalintomai) [inline components/apps_form/apps_form.tsx:106]: `unkown`->`unknown` (and below)

**Review by catalintomai — APPROVED** [review]

**Review by BenCookie95 — APPROVED** [review]

**Review by jfrerich — APPROVED** [review]

**Comment by larkox** [comment]
@DHaussermann I reviewed the logic of this, and here is a rundown:
- If there are valid errors in the form, the "invalid errors" are ignored
- If there are no valid errors in the form (neither field or general) but there is an "invalid error", the invalid error will show at the top of the form with the following message:
```
Received an error for an unknown field. Field name: `{field}`. Error:\n{error}
```

I will update the test app so you can test this.

**Comment by larkox** [comment]
@DHaussermann master of the test app has the needed command.

**Comment by DHaussermann** [comment]
Thanks @larkox with `/test valid open_markdown_form_modal_missing_error` I now see the error as expected. 

One last thing...  
>If there are no valid errors in the form (neither field or general) but there is an "invalid error", the invalid error will show at the top of the form with the following message:

This likely was the case before this PR but can this also be validated with the test-app?

**Comment by larkox** [comment]
The `/test valid open_markdown_form_modal` (or something like that) has errors everywhere, including a missing error.

**Comment by DHaussermann** [comment]
Thanks @larkox the explanation with bullets above makes sense to me now 👍

**Review by DHaussermann — APPROVED** [review]
Tested and passed
- Valid errors occur as expected 
- When no valid error is returned - invalid error is displayed as expected
- No issue found 

LGTM!

Test added to Zephyr to cover this. A separate PR is submitted to implement this mobile.

**Pull request merged**